tests/lib/newlib/heap_listener/src/main.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/libc_heap_listener.h>
+#include <zephyr.h>
+#include <ztest.h>
+
+#include <malloc.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+/* Function used by malloc() to obtain or free memory to the system */
+extern void *_sbrk(intptr_t count);
+
+static long current_heap_end(void)
+{
+	return (long) _sbrk(0);
+}
+
+static int heap_difference;
+
+static void heap_resized(void *old_heap_end, void *new_heap_end)
+{
+	heap_difference += ((char *) new_heap_end - (char *) old_heap_end);
```

Comment:
Cast unnecessary with `heap_difference` as `ptrdiff_t`.
```suggestion
	heap_difference += new_heap_end - old_heap_end;
```

Reply:
The case is necessary to avoid subtracting `void*`s which generates a warning when compiled with `-Wpointer-arith`

Reply:
Fair enough. void pointer arithmetic, although widely supported, is a GNU extension.